tests/DependencyInjection/SonataNewsExtensionTest.php

```diff
@@ -0,0 +1,113 @@
+     *
+     * @group legacy
+     * @expectedDeprecation The class %s is not defined or doesn't exist. This is tolerated now but will be forbidden in 4.0
+     */
+    public function testLoadWithoutTagWithoutCollection(): void
+    {
+        $this->load($this->getMinimalConfiguration());
+        $collector = DoctrineCollector::getInstance();
+
+        $this->assertEmpty($collector->getAssociations());
+    }
+
+    /**
+     * Test if the deprecation notice is trigger when the tag (or collection) class declaration is present.
+     * It shouldn't trigger a deprecation notice but doesn't break anything
+     * You should have 2 associations (Post, Comment).
+     * The Post model should have an associations with (Media, User, Collection, Tag).
+     */
+    public function testLoadWithTagWithCollection(): void
+    {
+        $minimalConfiguration = $this->getMinimalConfiguration();
+        $tagAndCollectionDeclaration = [
+                'class' => [
+                        'collection' => \stdClass::class,
+                        'tag' => \stdClass::class,
+                ],
+            ];
+        $this->load(array_merge($minimalConfiguration, $tagAndCollectionDeclaration));
+        $collector = DoctrineCollector::getInstance();
+        //assert our model have associations
+        $this->assertCount(2, $collector->getAssociations());
+        $postManyToOneAssociation = $collector->getAssociations()[Post::class]['mapManyToOne'];
+        //assert the post model has  3 many to one associations (user,media,collection)
+        $this->assertCount(3, $postManyToOneAssociation);
+        $postManyToManyAssociation = $collector->getAssociations()[Post::class]['mapManyToMany'];
+        //assert the post model has  1 many to many associations (tag)
+        $this->assertCount(1, $postManyToManyAssociation);
+        $postOneToManyAssociation = $collector->getAssociations()[Post::class]['mapOneToMany'];
+        //assert the post model has  1 one to many associations (comment)
+        $this->assertCount(1, $postOneToManyAssociation);
+    }
+
+    public function getMinimalConfiguration(): array
+    {
+        return [
+                'title' => 'Foo title',
+                'link' => '/foo/bar',
+                'description' => 'Foo description',
+                'salt' => 'pepper',
+                'comment' => [
+                    'notification' => [
+                        'emails' => ['email@example.org', 'email2@example.org'],
+                        'from' => 'no-reply@sonata-project.org',
+                        'template' => '@SonataNews/Mail/comment_notification.txt.twig',
+                    ],
+                ],
+                'class' => [
+                    'post' => Post::class,
+                    'comment' => Comment::class,
+                    'media' => Media::class,
+                    'user' => User::class,
+                ],
+            ];
+    }
+
+    /**
+     * {@inheritdoc}
```

Comment:
Please remove this

Reply:
I did it like in SonataMedia
https://github.com/sonata-project/SonataMediaBundle/blob/a190425f516d613d3ff136f0dd15b04feb322f3f/tests/DependencyInjection/SonataMediaExtensionTest.php#L282-L303

As I use this configuration twice, I wanted to be DRY. Why this is not a good idea?

Reply:
@core23 is not referring to the method, but to the phpdoc